static/app/views/settings/organizationTeams/teamSettings/index.tsx

```diff
@@ -11,125 +11,114 @@ import Form, {FormProps} from 'sentry/components/forms/form';
 import JsonForm from 'sentry/components/forms/jsonForm';
 import Panel from 'sentry/components/panels/panel';
 import PanelHeader from 'sentry/components/panels/panelHeader';
+import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
 import teamSettingsFields from 'sentry/data/forms/teamSettingsFields';
 import {IconDelete} from 'sentry/icons';
 import {t, tct} from 'sentry/locale';
-import {Organization, Team} from 'sentry/types';
+import {Team} from 'sentry/types';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
 import {normalizeUrl} from 'sentry/utils/withDomainRequired';
-import withOrganization from 'sentry/utils/withOrganization';
-import DeprecatedAsyncView from 'sentry/views/deprecatedAsyncView';
 import PermissionAlert from 'sentry/views/settings/project/permissionAlert';
 
-type Props = RouteComponentProps<{teamId: string}, {}> & {
-  organization: Organization;
+interface TeamSettingsProps extends RouteComponentProps<{teamId: string}, {}> {
   team: Team;
-};
-
-type State = DeprecatedAsyncView['state'];
-
-class TeamSettings extends DeprecatedAsyncView<Props, State> {
-  getTitle() {
-    return 'Team Settings';
-  }
-
-  getEndpoints() {
-    return [];
-  }
+}
 
-  handleSubmitSuccess: FormProps['onSubmitSuccess'] = (resp, _model, id) => {
-    const {organization} = this.props;
+function TeamSettings({team, params}: TeamSettingsProps) {
+  const organization = useOrganization();
+  const api = useApi({persistInFlight: true});
```

Comment:
Do we redirect immediately? Is that why we need this?

Reply:
because i don't want any requests to be cancelled either way

Reply:
why?

Reply:
do i want them to be cancelled?

Reply:
We usually want request to load data to be canceled on unmount, so that we don't do state updates on unmounted components.